src/main/java/gregtech/api/capability/impl/FuelRecipeLogic.java

```diff
@@ -207,21 +224,88 @@ public long getMaxVoltage() {
         return maxVoltage;
     }
 
+    protected double calculateFuelConsumptionMultiplier() {
+        return 1.0;
+    }
+
     protected int calculateFuelAmount(FuelRecipe currentRecipe) {
-        return currentRecipe.getRecipeFluid().amount * getVoltageMultiplier(getMaxVoltage(), currentRecipe.getMinVoltage());
+        return (int) (currentRecipe.getRecipeFluid().amount * getVoltageMultiplier(getMaxVoltage(), currentRecipe.getMinVoltage()) * calculateFuelConsumptionMultiplier());
+    }
+
+    public int getFuelAmount() {
+        return calculateFuelAmount(this.previousRecipe);
+    }
+
+    protected double calculateRecipeDurationMultiplier() {
+        return 1.0;
     }
 
     protected int calculateRecipeDuration(FuelRecipe currentRecipe) {
-        return currentRecipe.getDuration();
+        return (int) (currentRecipe.getDuration() * calculateRecipeDurationMultiplier());
+    }
+
+    public int getRecipeDuration() {
+        return calculateRecipeDuration(this.previousRecipe);
+    }
+
+    /**
+     * this method aim to tell if the current recipe has ended in order to check for a new recipe see {@link #isActive}
+     * or {@link #isWorkingEnabled} if you want to know if the machine is running.
+     *
+     * @return true if no recipe is running false otherwise
+     */
+    public boolean hasRecipeEnded() {
+        return this.recipeDurationLeft <= 0;
+    }
+
+    public boolean canProduceEnergy() {
+        return !hasRecipeEnded();
+    }
+
+    public boolean canConsumeFuel() {
+        return !hasRecipeEnded();
+    }
+
+    /**
+     * Compute a multiplier used by default in {@link #startRecipe(FuelRecipe, int, int)}  startRecipe} each time the recipe start.
+     * fast changing efficiency could be implemented in {@link #calculateDynamicEnergyEfficiency() calculateDynamicEnergyEfficiency} for better responsiveness of the changes.
+     *
+     * @return power multiplier
+     */
+    protected double calculateStaticEnergyEfficiency() {
```

Comment:
What is the point of this method? What is the point of having "static" energy efficiency in general?
What can you achieve using "static" energy efficiency that you cannot achieve by returning constant value in calculateDynamicEnergyEfficiency?

Reply:
once again go and read the comment in #1604 I pointed that in the additional context.

Reply:
Still makes no sense to introduce method for something so niche and resolvable by other means

Reply:
Once again, your comment on that PR does not advocate adding this method
You gave only a single argument, which is "performance", but performance difference is literally nonexistent in that particular case, and it makes no sense to introduce additional useless method for it

Reply:
>You gave only a single argument, which is "performance", but performance difference is literally nonexistent in that particular case, and it makes no sense to introduce additional useless method for it

okay in this case I don't see any reason keeping those methods separated

Reply:
>but performance difference is literally nonexistent in that particular case, and it makes no sense to introduce additional useless method for it

just asking but if you have a relatively  expensive computation needed only each recipe start and the recipe is very long could it be worth to keep things splitted ?